lib/jekyll/filters.rb

```diff
@@ -68,22 +68,33 @@ def slugify(input, mode = nil)
     end
 
     # Format a date in short format e.g. "27 Jan 2011".
+    # It allows to use ordinal format as well, in both UK
+    # (e.g. "27th Jan 2011") and US ("e.g. Jan 27th, 2011") formats.
+    # UK format is the default.
     #
     # date - the Time to format.
+    # type - if "ordinal" the returned String will be in ordinal format
+    # style - if "US" the returned String will be in US format.
+    #         Otherwise it will be in UK format.
     #
     # Returns the formatting String.
-    def date_to_string(date)
-      time(date).strftime("%d %b %Y")
+    def date_to_string(date, type = nil, style = nil)
+      generic_date_to_string(date, "b", type, style)
     end
 
     # Format a date in long format e.g. "27 January 2011".
+    # It allows to use ordinal format as well, in both UK
+    # (e.g. "27th January 2011") and US ("e.g. January 27th, 2011") formats.
+    # UK format is the default.
     #
-    # date - The Time to format.
+    # date - the Time to format.
+    # type - if "ordinal" the returned String will be in ordinal format
+    # style - if "US" the returned String will be in US format.
+    #         Otherwise it will be in UK format.
     #
     # Returns the formatted String.
-    def date_to_long_string(date)
-      return date if date.to_s.empty?
-      time(date).strftime("%d %B %Y")
+    def date_to_long_string(date, type = nil, style = nil)
+      generic_date_to_string(date, "B", type, style)
```

Comment:
Could you convert the magic string `B` into a constant? Same for `b` as well. Something like:
```ruby
LONG_MONTH_FORMAT = 'B'
SHORT_MONTH_FORMAT = 'b'
```

provides much more expressiveness and clarity in the code than the magic string.

Reply:
I changed it to @ashmaroli suggestion:

> instead of passing b and B as parameters, how about passing %b and %B ?

Do you think it is already clear?

I think with that is already clear that is the use of the strftime method and IMO there is no need to add two variables for this. :smile: